src/services/EcoBridge/Xdai/XdaiBridge.reducer.ts

```diff
@@ -0,0 +1,139 @@
+import { createSlice, EntityState, PayloadAction } from '@reduxjs/toolkit'
+import { TokenList } from '@uniswap/token-lists'
+
+import { BridgeTransactionStatus } from '../../../state/bridgeTransactions/types'
+import { BridgeDetails, BridgingDetailsErrorMessage, SyncState, XdaiBridgeList } from '../EcoBridge.types'
+import { xdaiBridgeTransactionAdapter } from './XdaiBridge.adapter'
+import { XdaiBridgeTransaction, XdaiMessage } from './XdaiBridge.types'
+
+interface XdaiBridgeState {
+  bridgingDetails: BridgeDetails
+  bridgingDetailsStatus: SyncState
+  lastMetadataCt: number
+  listsStatus: SyncState
+  lists: { [id: string]: TokenList }
+  transactions: EntityState<XdaiBridgeTransaction>
+}
+
+const initialState: XdaiBridgeState = {
+  transactions: xdaiBridgeTransactionAdapter.getInitialState({}),
+  bridgingDetails: {},
+  bridgingDetailsStatus: SyncState.IDLE,
+  lists: {},
+  listsStatus: SyncState.IDLE,
+  lastMetadataCt: 0,
+}
+
+const createXdaiSlice = (bridgeId: XdaiBridgeList) =>
+  createSlice({
+    name: bridgeId,
+    initialState,
+    reducers: {
+      setBridgeDetails: (state, action: PayloadAction<BridgeDetails>) => {
+        const { gas, fee, estimateTime, receiveAmount, requestId } = action.payload
+
+        //(store persist) crashing page without that code
+        if (!state.bridgingDetails) {
+          state.bridgingDetails = {}
+        }
+
+        if (requestId !== state.lastMetadataCt) {
+          if (state.bridgingDetailsStatus === SyncState.FAILED) return
+          state.bridgingDetailsStatus = SyncState.LOADING
+          return
+        } else {
+          state.bridgingDetailsStatus = SyncState.READY
+        }
+
+        state.bridgingDetails.gas = gas
+
+        state.bridgingDetails.fee = fee
+
+        state.bridgingDetails.estimateTime = estimateTime
+
+        state.bridgingDetails.receiveAmount = receiveAmount
```

Comment:
```suggestion
        if (requestId !== state.lastMetadataCt) {
          if(state.bridgingDetailsStatus !== SyncState.FAILED){
               state.bridgingDetailsStatus = SyncState.LOADING   
            }       
        } else {
          state.bridgingDetailsStatus = SyncState.READY
          state.bridgingDetails.gas = gas
          state.bridgingDetails.fee = fee
          state.bridgingDetails.estimateTime = estimateTime
          state.bridgingDetails.receiveAmount = receiveAmount
        }
```

Is it same?

Reply:
fixed [f669193](https://github.com/levelkdev/swapr-dapp/pull/1089/commits/f66919383c50bce1cf19d4366172760f1055f43d)